lib/Parse/Lexer.cpp

```diff
@@ -622,6 +622,13 @@ static bool isLeftBound(const char *tokBegin, const char *bufferBegin) {
   // The first character in the file is not left-bound.
   if (tokBegin == bufferBegin) return false;
 
+  if (tokBegin - 1 != bufferBegin) {
+    if (tokBegin[-2] == '*' && tokBegin[-1] == '/') {
+      // End of a slash-star comment, which we treat as whitespace.
+      return false;
+    }
+  }
+
```

Comment:
This logic looks correct, but for better efficiency, please put this into the default case in the switch below.  That way it isn't run in the common case where a token is preceded by whitespace.